server/enclave/ecalls.cpp

```diff
@@ -120,45 +123,64 @@ void enclave_modelaggregator(int tid) {
     int slice_length = 1 + ((g_vars_to_aggregate.size() - 1) / NUM_THREADS);
 
     // Slice the vector depending on thread ID
-    auto first = g_vars_to_aggregate.begin() + tid * slice_length;
-    auto last = g_vars_to_aggregate.begin() + min((int) g_vars_to_aggregate.size(), (tid + 1) * slice_length);
-    vector<string> vars_slice(first, last);
+    int i = tid * slice_length;
+    int j = min((int) g_vars_to_aggregate.size(), (tid + 1) * slice_length);
 
     // Each thread iterates through a portion of all weights names received by the clients.
-    for (string v_name : vars_slice) {
-        double iters_sum = 0;
-        vector<double> updated_params_at_var(g_old_params[v_name]);
+    for (; i < j; i++) {
+        string v_name = g_vars_to_aggregate[i];
+        float iters_sum = 0;
+
         // For each accumulator, we find the vector of the current weight and
         // multiple all of it's elements by local iterations. We keep a running
         // sum of total iterations and a vector of all weights observed.
-        for (map<string, vector<double>> acc_params : g_accumulator) {
+        for (int k = 0; k < g_accumulator.size(); k++) {
+            map<string, vector<float>> acc_params = g_accumulator[k];
             if (acc_params.find(v_name) == acc_params.end()) { // This accumulator doesn't have the given variable
                 continue;
             }
 
             // Each params map will have an additional key "_contribution" to hold the number of local iterations.
-            double n_iter = acc_params["_contribution"][0];
+            float n_iter = acc_params["_contribution"][0];
             iters_sum += n_iter;
 
-            // Multiple the weights by local iterations.
-            vector<double>& weights = acc_params[v_name];
-            if (updated_params_at_var.size() != weights.size()) {
+            if (g_old_params[v_name].size() != acc_params[v_name].size()) {
                 std::cout << "Error! Unequal sizes" << std::endl;
             }
 
-            for (int i = 0; i < weights.size(); i++) {
-                updated_params_at_var[i] += weights[i] * n_iter;
+            __m256 iters_sum_slice;
+            if (k == g_accumulator.size() - 1 && iters_sum > 0) {
+                const float iters_sum_arr[8] = {iters_sum, iters_sum, iters_sum, iters_sum,
+                                                    iters_sum, iters_sum, iters_sum, iters_sum};
+                iters_sum_slice = _mm256_loadu_ps(iters_sum_arr);
```

Comment:
I think you can use `_mm256_broadcast_ss(&iters_sum)` instead of `_mm256_loadu_ps`. That way you don't have to first create an array. I tested this and it builds/runs, but you may want to check correctness.